### PR TITLE
feat: add basic layout frame

### PR DESCRIPTION
**Pull request by turbodog03**
# I add a basic layout frame for casdoor. Includes collapsible sidebar, header, context and footer.

## changes:
* I moved the navigation menu from header to sidebar ( so we can show other electments on header and homepage.
* Move the muti-language select box from footer to header and use a select for selections. #165
* Add icons for each menu items.
* Add breadcrumb( to be improved
* Add siteName and logoPath to conf.js, to enable users to customerize it
* Online demo: http://39.106.18.82/
## to be improved
* Develop a dynamic breadcrumb
* Make a new homepage, to show backend datas.
* Optimize the components
* Add more features if needed

### CODE REVIEW

**Comment by turbodog03** [comment]
please review @nodece

File: web/src/basic/HomePage.js [hunk 1]
```diff
@@ -30,18 +31,43 @@ class HomePage extends React.Component {
...
+          organizer: i18next.t("account:Settings for your account"),
+        },
       ];
     }
-
-    for (let i = 0; i < items.length; i ++) {
+    // items =[
+    // {link: "https://casdoor.org/zh/docs/basic/basic-concept-introduction", name: i18next.t("general:Basic concepts "), organizer: i18next.t("general:Welcome to casdoor! Here are some Basic concepts to help you getting start.")},
+    // {link: "https://casdoor.org/zh/docs/basic/basic-concept-introduction", name: i18next.t("general:Add an provider "), organizer: i18next.t("general:Casdoor uses providers to provide third-party services for the platform. let's learn how to add your first provider.")},
+    // ];
```
Comment (nodece) [inline web/src/basic/HomePage.js:69]: Remove commented code.

File: web/src/basic/HomePage.js [hunk 2]
```diff
@@ -50,62 +76,77 @@ class HomePage extends React.Component {
...
         </div>
-      )
+      );
     }
   }
 
   render() {
     return (
       <div>
+        {/* <span className= "contentTitle">
```
Comment (nodece) [inline web/src/basic/HomePage.js:139]: Remove the commented out code.

File: web/src/basic/SingleCard.js [hunk 3]
```diff
@@ -29,47 +29,68 @@ class SingleCard extends React.Component {
...
     };
 
     return (
-      <Card.Grid style={gridStyle} onClick={() => Setting.goToLinkSoft(this, link)}>
-        <img src={logo} alt="logo" height={60} style={{marginBottom: '20px'}}/>
-        <Meta
-          title={title}
-          description={desc}
-        />
+      // <Card.Grid style={gridStyle} onClick={() => Setting.goToLinkSoft(this, link)}>
```
Comment (nodece) [inline web/src/basic/SingleCard.js:38]: Remove the commented out code.

File: web/src/basic/SingleCard.js [hunk 4]
```diff
@@ -29,47 +29,68 @@ class SingleCard extends React.Component {
...
+      <Card.Grid
+        style={gridStyle}
+        onClick={() => Setting.goToLinkSoft(this, link)}
+      >
+        {/* <img src={logo} alt="logo" height={60} style={{marginBottom: '20px'}}/> */}
```
Comment (nodece) [inline web/src/basic/SingleCard.js:43]: Remove the commented out code.

File: web/src/component/SelectLanguageBox.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    );
+
+    return (
+      <Dropdown overlay={menu}>
+        <span className="language_box" />
+      </Dropdown>
+    );
+  }
+
+  // render() {
```
Comment (nodece) [inline web/src/component/SelectLanguageBox.js:52]: Remove the commented out code.

File: web/src/component/SelectLanguageBox.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+class SelectLanguageBox extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      classes: props,
+    };
+  }
+
+  onClick = function (e) {
```
Comment (nodece) [inline web/src/component/SelectLanguageBox.js:28]: So it looks cleaner:
```
onClick(e) {
// todo
}
```

File: web/src/layout/index.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+import * as Setting from "../Setting.js";
+import CustomGithubCorner from "../CustomGithubCorner";
+import { Link, withRouter } from "react-router-dom";
+import i18next from "i18next";
+import { Nav } from "../component/breadcrumb";
+import SizeContext from "antd/lib/config-provider/SizeContext";
+import * as conf from "../common/Conf.js";
+const { Header, Footer, Sider, Content } = Layout;
+
+// 引入子菜单组件
```
Comment (nodece) [inline web/src/layout/index.js:13]: Remove this comment.

File: web/src/layout/index.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+const SubMenu = Menu.SubMenu;
+
+export class BasicSider extends Component {
+  state = {
+    collapsed: false,
+  };
+
+  onCollapse = (collapsed) => {
+    console.log(collapsed);
```
Comment (nodece) [inline web/src/layout/index.js:22]: Remove `console`.

File: web/src/layout/index.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    this.setState({ collapsed });
+  };
+
+  render() {
+    const { collapsed } = this.state;
+    console.log(this.props.path);
```
Comment (nodece) [inline web/src/layout/index.js:28]: Remove `console`

**Review by nodece — CHANGES_REQUESTED** [review]

**Review by nodece — APPROVED** [review]
LGTM

**Comment by hsluoyz** [comment]
@turbodog03 I don't think this layout is anything better than our existing look.

1. I don't like the sidebar. It takes more space of the width.
2. The UI look is neither fancy or professional. It is actually very similar to the Casbin-dashabord project which has failed one year ago (https://github.com/casbin/casbin-dashboard) . 

I can't name a strong reason to accept this huge PR as a whole. Maybe some minor improvements are welcome, like language select, path conf, etc.

image

**Pull request closed without merging**